docs/ref/models/instances.txt

```diff
@@ -197,6 +197,23 @@ all of the instance's fields when a deferred field is reloaded::
                     fields = fields.union(deferred_fields)
             super().refresh_from_db(using, fields, **kwargs)
 
+The ``from_queryset`` parameter allows to use a different queryset than the one
+created from ``_base_manager``. It gives you more control over how the model is
+reloaded. When your model uses soft deletion you can make ``refresh_from_db``
+take that into account::
+
+    obj.refresh_from_db(from_queryset=Model.active_objects.all())
+
+You can cache related models that otherwise would be cleared from the reloaded
+instance::
+
+    obj.refresh_from_db(from_queryset=Model.objects.select_related('related_field')
+
+You can lock the row until the end of transaction before reloading model's values::
+
+    obj.refresh_from_db(from_queryset=Model.objects.select_for_()
```

Comment:
This last example looks accidentally truncated, I _assume_ it's something like:
```suggestion
    obj.refresh_from_db(from_queryset=Model.objects.select_for_update())
```
give or take arguments for `select_for_update` ...?